certgen/csr_w_ed25519_example.c

```diff
@@ -0,0 +1,100 @@
+#include <wolfssl/options.h>
+#include <wolfssl/wolfcrypt/settings.h>
+#include <wolfssl/wolfcrypt/ed25519.h>
+#include <wolfssl/wolfcrypt/asn_public.h>
+
+#define MAX_TEMP_SIZE 1024
+
+int main(void)
+{
+#if !defined(HAVE_ED25519) || !defined(WOLFSSL_CERT_REQ) || \
+    !defined(WOLFSSL_CERT_GEN)
+  printf("The csr_w_ed25519_example will not work unless wolfSSL is\n"
+         "configured with the following settings:\n"
+         "--enable-ed25519 --enable-certreq --enable-certgen\n");
+  return 0;
+#else
+    int ret;
+    ed25519_key key;
+    WC_RNG rng;
+    Cert req;
+    byte der[MAX_TEMP_SIZE], pem[MAX_TEMP_SIZE];
+    int  derSz, pemSz;
+
```

Comment:
Please add a memset on the `key` and `rng` so the error case will handle these correctly if not initialized.